app/utils.py

```diff
@@ -91,250 +109,335 @@ def fetch_and_store_players():
-            if Player.player_exists(player_id):
-                # Add the player if they don't exist
-                # Add position and how_aquired into roster, and position, age, height, weight, draft into player
-
-                ##position = player.get('POSITION', 'Unknown')
-                team_name = player['TeamID']
-
-                #Player.update_player(player_id,player_name,player_number,position,weight, born_date,age,exp,school)
-                
-                #print(f"Added new player: {player_name} (ID: {player_id})(team_name(ID): {team_name})(position: {position})")
-            else:
-                print(f"PlayerID: {player_id} Name: {player_name} does not exist, adding to DB \n ")
-                Player.add_player(player_id, player_name,position, weight,born_date, age,exp,school, "Unknown")
-                
-            # Add player to the team's roster in the database
-            team_obj.add_to_roster(player_id=player_id, player_name=player_name, player_number=player_number,position=position, how_acquired=how_acquired, season = season)
-    
-    #update_player_teams()
-    print("All current NBA teams and their rosters have been successfully stored.")
+        team_obj = Team.add_team(name=team_name, abbreviation=team_abbreviation)
+        logger.info(
+            "Added %s with ID: %s. Sleeping for 15 seconds.", team_name, team_id
+        )
+        time.sleep(15)
+
+        try:
+            # Fetch the roster for this team
+            team_roster_data = commonteamroster.CommonTeamRoster(
+                team_id=team_id, timeout=600
+            ).get_normalized_dict()
+            team_roster = team_roster_data["CommonTeamRoster"]
+
+            logger.info(
+                "Fetched roster for team %s with %s players.",
+                team_name,
+                len(team_roster),
+            )
+
+            # Add each player in the roster to the team
+            for player in team_roster:
+                time.sleep(1)
+                player_id = player["PLAYER_ID"]
+                player_name = player["PLAYER"]
+                player_number = player["NUM"]  # Jersey number
+                position = player["POSITION"]
+                how_acquired = player["HOW_ACQUIRED"]
+                season = player["SEASON"]
+
+                # Player Data
+                weight = player.get("WEIGHT", 0)
+                born_date = player.get("BIRTH_DATE", "Unknown")
+                age = player.get("AGE", 0)
+                exp = player.get("EXP", "0")
+                if exp == "R":
+                    exp = 0
+                elif exp.isdigit():
+                    exp = int(exp)
+                else:
+                    exp = 0
+
+                school = player.get("SCHOOL", "Unknown")
+
+                logger.info("Processing player %s (ID: %s).", player_name, player_id)
+                parsed_age = age_parser(age)
+                Player.add_player(
```

Comment:
we cant remove this if check for player_exists(). It will error out or add duplicates to the player database. In your code fetching rosters for multiple years would result in duplicate Player.add_player()

Reply:
add if statement for if player dne,  then add player else, skip and move on to add_to_roster

Reply:
added if not exists condition